manual.typ

```diff
@@ -1415,3 +1374,187 @@ tree.tree(data, content: (padding: .1), direction: "right",
 A tree node is an array of nodes. The first array item represents the
 current node, all following items are direct children of that node.
 The node itselfes can be ot type `content` or `dictionary` with a key `content`.
+
+== Plot
+
+The library `plot` of CeTZ allows plotting 2D data as linechart.
+
+=== Plot <plot-function>
+
+The `plot` function is an environment for plotting data.
+
+Note that different axis-styles can show different axes. The "school-book" style shows only axis "x" and "y", while the "scientific" style can show "x2" and "y2", if set (if unset, "x2" mirrors "x" and "y2" mirrors "y"). Other axes (e.G. "my-axis") work, but no ticks or labels will be shown.
+
+```typc
+plot(size: (width, height),
+     axis-style: "scientific"
+     ..options,
+     body)
+```
+
+The ticks option dictionary supports the following keys:
+#def-arg("size", "a",
+  [Size of the plot])
+#def-arg("axis-style", "s?",
+  [Axis style, either "scientific" or "school-book"])
+#def-arg("..options", "any",
+  [Axis options in the form `<axis-name>-<option>`. The possible options are:
+  #box[
+  / label : The axis label
+  / min: Axis min. value
+  / max: Axis max. value
+  / ticks: List of tick values or value/label tuples
+  / tick-step: Tick step or `none`
+  / minor-tick-step: Minor tick step or `none`
+  / decimals: Number of tick label decimal digits
+  / unit: Tick label suffix
+  ]
+])
```

Comment:
Could the axes use a similar input to styling? As in nested dictionaries:
```
plot(
  axes: (
    x: (
      label: "X",
      ...
     ),
    y: ...
  )
)
```
You could then allow options to be applied to both axes by allowing them in the base of the `axes` dict or through a `default` dict.
```
plot(
  axes: (
   // Applies to all axes
  ticks: ...
   // or if you want to keep axis names clear
   default: (
      ticks: ...
    ),
   // An axis
    y: ...
  )
)
```
This might even allow them to be set through `set-style`?t

Reply:
You might also want to group tick options like step, minor-step, values etc.

Reply:
I am not sure if that is good, as it comes with no real benefit.